File: 01-BigO/bigO_examples.js
// "O большое" используентся
// описания производительности алгоритмов 
// (времени выполнения или использования памяти) 
// в зависимости от размера входных данных.

function addUpTo(n) {
    // значение O для функции: 
    // O(n)
    let total = 0;
    for (let i = 0; i <= n; i++) {
        total += i;
    }
    return total;
}

console.log(addUpTo(6))


function addUpTo(n) {
    // значение O для функции: 
    // O(1)
    return n * (n + 1) / 2;
}

console.log(addUpTo(6))


function printAllPairs(n) {
    // значение O для функции: 
    // O(n**2)
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            console.log(i, j)
        }
    }
}

printAllPairs(3)

File: 01-BigO/bigO_examples.py
# "O большое" используется
# для описания производительности алгоритмов 
# (времени выполнения или использования памяти) 
# в зависимости от размера входных данных.

def add_up_to(n):
    # значение O для функции: 
    # O(n)
    total = 0
    for i in range(n + 1):
        total += i
    return total

print(add_up_to(6))


def add_up_to_constant(n):
    # значение O для функции: 
    # O(1)
    return n * (n + 1) // 2

print(add_up_to_constant(6))


def print_all_pairs(n):
    # значение O для функции: 
    # O(n**2)
    for i in range(n):
        for j in range(n):
            print(i, j)

print_all_pairs(3)
